scripts/install_xcpng.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python3
+import argparse
+import atexit
+import os
+import requests
+import sys
+import tempfile
+import time
+try:
+    sys.path.append(f"{os.path.abspath(os.path.dirname(__file__))}/..")
+except Exception:
+    pass
+from lib.commands import ssh, scp, SSHCommandFailed
+from lib.common import wait_for, is_uuid
+from lib.host import host_data
+from lib.pool import Pool
+from lib.vm import VM
+from packaging import version
+
+def pxe_address():
+    try:
+        from data import DEF_PXE_IP
+        address_pxe = DEF_PXE_IP
+    except ImportError:
+        raise Exception('No address for the PXE server in the data.py')
+    return address_pxe
+
+def generate_boot_conf(tmp_directory, installer):
+    bootfile = f'{tmp_directory}/boot.conf'
+    fb = open(bootfile, 'w')
+    x = f"""answerfile=custom
+installer={installer}
+is_default=1
+"""
+    fb.write(x)
+    fb.close()
+
+def generate_answerfile(tmp_directory, installer, hostname_or_ip, type, hdd):
+    pxe = pxe_address()
+    password = host_data(hostname_or_ip)['password']
+    tempstream = os.popen(f"mkpasswd -m SHA-512 '{password}'")
+    encrypted_password = tempstream.read()
+    answerfile = f'{tmp_directory}/answerfile.xml'
+    fa = open(answerfile, 'w')
```

Comment:
Same here. `with open`. :)

Reply:
```py
    with open(f'{tmp_directory}/answerfile.xml', 'w') as fa:
       ...
```

Reply:
`with open` is a context manager. You can remove calls to `file.write(...)` and `file.close()` with it.